src/apps/datasets/models.py

```diff
@@ -60,12 +63,64 @@ def save(self, *args, **kwargs):
             self.name = f"{self.created_by.username} - {self.type}"
         return super().save(*args, **kwargs)
 
+    @property
+    def in_use(self):
+        from tasks.models import Task
+        tasks = Task.objects.filter(
+            Q(ingestion_program=self) |
+            Q(input_data=self) |
+            Q(reference_data=self) |
+            Q(scoring_program=self)
+        )
+        phases_from_tasks = [phase for task in tasks for phase in task.phases.all()]
```

Comment:
why are we querying tasks to filter back up to competitions when we can query competitions based on tasks?